packages/components/src/components/files/WebDav.ts

```diff
@@ -0,0 +1,283 @@
+
+import { Map, Record } from 'immutable';
+import { ActionURL, Ajax, Utils } from '@labkey/api';
+import {DEFAULT_FILE, FileAttachmentFormModel, IFile} from "./models";
+
+export class WebDavFile extends Record(DEFAULT_FILE) implements IFile {
+    contentLength: number;
+    created: string;
+    createdBy: string;
+    createdById: number;
+    dataFileUrl: string;
+    description: string;
+    downloadUrl: string;
+    href: string;
+    id: string;
+    iconFontCls: string;
+    isCollection: boolean;
+    isLeaf: boolean;
+    lastModified: string;
+    name: string;
+    propertiesRowId?: number;
+
+    constructor(values?: { [key: string]: any }) {
+        super(values);
+    }
+
+    /**
+     * values is presumed to be an object returned from the webdav controller.  It has several fields we are not
+     * currently using in the application.  We pick out the ones we want and make their names more standard.
+     * N.B.  The "size" field is particularly problematic since records already have a size field, so don't use that
+     * (contentlength seems to have the same value anyway).
+     * @param values
+     */
+    static create(values): WebDavFile {
+        let props = {};
+
+        if (values['collection'])
+            props['isCollection'] = values['collection'];
+        if (values['contentlength'])
+            props['contentLength'] = values['contentlength'];
+        if (values['createdby'])
+            props['createdBy'] = values['createdby'];
+        if (values['creationdate'])
+            props['created'] = values['creationdate'];
+        if (values['dataFileUrl'])
+            props['dataFileUrl'] = values['dataFileUrl'];
+        if (values['description'])
+            props['description'] = values['description'];
+        if (values['href']) {
+            props['href'] = values['href'];
+            props['downloadUrl'] = props['href'] + "?contentDisposition=attachment";
+        }
+        if (values['iconFontCls'])
+            props['iconFontCls'] = values['iconFontCls'];
+        if (values['id'])
+            props['id'] = values['id'];
+        if (values['lastmodified'])
+            props['lastModified'] = values['lastmodified'];
+        if (values['leaf'])
+            props['isLeaf'] = values['leaf'];
+        if (values['text'])
+            props['name'] = values['text'];
+        return new WebDavFile(props);
+    }
+}
+
+export interface WebDavContainer extends FileAttachmentFormModel {
+    containerPath: string
+    isLoading: boolean
+    uploadedFiles: Map<string, WebDavFile>
+}
+
+export class WebDav extends Record({
```

Comment:
This class should not extend record. It's not even used as a record really. Drop the extends part, drop the constructor that's not used anywhere.